source/simulator/parameters.cc

```diff
@@ -858,6 +888,24 @@ namespace aspect
                    ExcMessage("Invalid input parameter file: Too many entries in List of normalized fields"));
     }
     prm.leave_subsection ();
+
+
+    // now also get the parameters related to material model averaging
+    prm.enter_subsection ("Material model");
+    {
+      material_averaging_for_linear_systems
+        = MaterialModel::MaterialAveraging::parse_averaging_operation_name
+          (prm.get ("Material averaging for linear systems"));
+      material_averaging_for_preconditioners
+        = MaterialModel::MaterialAveraging::parse_averaging_operation_name
+          (prm.get ("Material averaging for preconditioners"));
+
+      // implement the special rule that if the linear systems use some
+      // averaging, then the preconditioner needs to use the same
+      if (material_averaging_for_linear_systems != MaterialModel::MaterialAveraging::none)
```

Comment:
Can you explain to me why this is necessary? This logic doesn't seem to be documented in the parameter description, which might be confusing to the user.

You could remove  "Material averaging for linear systems" and just add a bool "also use averaging for linear system", right?